# **Projet** 

"Les maladies cardiaques, également appelées maladies cardiovasculaires, constituent un terme générique désignant les affections et troubles qui touchent le cœur et le système circulatoire. Elles représentent l’une des principales causes d’invalidité dans le monde. Comme le cœur est l’un des organes les plus vitaux du corps humain, ses maladies ont également un impact sur d’autres organes et parties du corps.
Il existe plusieurs types et formes de maladies cardiaques. Les plus courantes provoquent un rétrécissement ou une obstruction des artères coronaires, un dysfonctionnement des valves cardiaques, une dilatation du cœur, ainsi que d’autres anomalies pouvant conduire à une **insuffisance cardiaque** ou à une **crise cardiaque**.*"

In [1]:
import pandas as pd
data = pd.read_csv('heart.csv')
print('Shape of the data is ', data.shape)

Shape of the data is  (303, 14)


In [6]:
data.head()

âge  sexe  type_douleur_poitrine  tension_repos  cholestérol  \
0   63     1                      3            145          233   
1   37     1                      2            130          250   
2   41     0                      1            130          204   
3   56     1                      1            120          236   
4   57     0                      0            120          354   

   glycémie_jeune  électrocardiogramme_repos  fréquence_cardiaque_max  \
0               1                          0                      150   
1               0                          1                      187   
2               0                          0                      172   
3               0                          1                      178   
4               0                          1                      163   

   angine_exercice  dépression_ST_exercice  pente_ST  vaisseaux_colorés  thal  \
0                0                     2.3         0                  0     1   
1                0                     3.5         0                  0     2   
2                0                     1.4         2                  0     2   
3                0                     0.8         2                  0     2   
4                1                     0.6         2                  0     2   

   maladie_cœur  
0             1  
1             1  
2             1  
3             1  
4             1

In [9]:
data.dtypes

âge                            int64
sexe                           int64
type_douleur_poitrine          int64
tension_repos                  int64
cholestérol                    int64
glycémie_jeune                 int64
électrocardiogramme_repos      int64
fréquence_cardiaque_max        int64
angine_exercice                int64
dépression_ST_exercice       float64
pente_ST                       int64
vaisseaux_colorés              int64
thal                           int64
maladie_cœur                   int64
dtype: object

1.1 Dictionnaire des données <a class="anchor" id="1.1"></a>

âge : âge du patient (en années)

sexe : sexe du patient

1 = homme

0 = femme

type_douleur_poitrine : type de douleur thoracique

Valeur 0 : angine typique

Valeur 1 : angine atypique

Valeur 2 : douleur non angineuse

Valeur 3 : asymptomatique

tension_repos : tension artérielle au repos (en mm Hg à l’admission à l’hôpital)

cholestérol : taux de cholestérol sérique (en mg/dl)

glycémie_jeune : glycémie à jeun (> 120 mg/dl)

1 = vrai

0 = faux

électrocardiogramme_repos : résultats de l’électrocardiogramme au repos

Valeur 0 : normal

Valeur 1 : anomalies de l’onde ST-T (inversion de l’onde T et/ou élévation ou dépression du segment ST > 0,05 mV)

Valeur 2 : hypertrophie ventriculaire gauche probable ou certaine (critères d’Estes)

fréquence_cardiaque_max : fréquence cardiaque maximale atteinte à l’effort

angine_exercice : angine de poitrine provoquée par l’effort

1 = oui

0 = non

dépression_ST_exercice : dépression du segment ST induite par l’effort par rapport au repos

pente_ST : pente du segment ST lors de l’effort maximal

Valeur 0 : montante

Valeur 1 : plate

Valeur 2 : descendante

vaisseaux_colorés : nombre de vaisseaux majeurs (0 à 3) colorés par fluoroscopie

thal : résultat du test au thallium

0 = erreur (dans le jeu de données original, 0 correspond à des valeurs manquantes)

1 = défaut fixe

2 = normal

3 = défaut réversible

maladie_cœur (étiquette cible) : présence d’une maladie cardiaque

0 = pas de maladie

1 = maladie détectée